pilot/pkg/xds/sds.go

```diff
@@ -64,7 +64,7 @@ func sdsNeedsPush(updates model.XdsUpdates) bool {
 	if len(updates) == 0 {
 		return true
 	}
-	if len(model.ConfigNamesOfKind(updates, gvk.Secret)) > 0 {
+	if model.ConfigsHaveKind(updates, gvk.Secret) || model.ConfigsHaveKind(updates, gvk.ReferencePolicy) {
```

Comment:
Wait, IIUC ReferencePolicy can reference any resource, how could you say it will influence sds

Reply:
I did so because now we only supports reference policies for gateway->secret, but I think we will implement more policies

Fixed